### DISCUSSION TITLE
Setting translation keys in HTML format

### DISCUSSION THREAD

**Discussion by rubenwardy** [WeblateOrg/weblate#6145] [Q&A]
Hi,

I'm using weblate to translate the HTML document found here: https://gitlab.com/rubenwardy/renewedtab/-/blob/master/src/app/locale/store/en.html

Unfortunately, weblate constantly breaks when updating translations from the en.html source because it uses keys based on the order and type of elements. This results in the translations not matching the source

I'd like to be able to supply fixed keys for DOM elements. I tried using `id="somekey"` on the containing element, but it still uses the old key. Alternatively, the key should be based on the textContent not the element type and position

My instance is here: https://weblate.rubenwardy.com/.
Edit: I've had to delete the project as it keeps corrupting the files.
It's an open source project, but probably isn't big enough to qualify for hosted weblate

**Comment by nijel** [comment]
Using `id` as another way to identify string location could probably be added to https://github.com/translate/translate/blob/master/translate/storage/html.py

**Reply by rubenwardy** [reply]
Thanks for the suggestion, I've created an issue here: https://github.com/translate/translate/issues/4369
